flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -81,7 +86,26 @@ private static Config defaultConfig() {
    * @return a {@link Result} with the value and throwable (if thrown)
    */
   public static <T> Result<T> runTask(Task<T> task, Config config) {
-    return new Result<>(new FloRunner<T>(config).run(task));
+    return new Result<>(new FloRunner<T>(config).run(task), loadTerminationHooks());
+  }
+
+  private static Iterable<TerminationHook> loadTerminationHooks() {
+    final ServiceLoader<TerminationHookFactory> factories =
+        ServiceLoader.load(TerminationHookFactory.class);
+
+    return StreamSupport
+        .stream(
+            Spliterators.spliteratorUnknownSize(factories.iterator(), Spliterator.ORDERED), false)
+        .filter(Objects::nonNull)
+        .map(factory -> {
+          try {
+            return factory.create();
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        })
+        .filter(Objects::nonNull)
```

Comment:
This also hides potential issue. Let it fail instead of swallowing? I mean we don't need to check null.